WordPress/Sniffs/Utils/I18nTextDomainFixerSniff.php

```diff
@@ -0,0 +1,725 @@
+
+namespace WordPress\Sniffs\Utils;
+
+use WordPress\AbstractFunctionParameterSniff;
+use PHP_CodeSniffer_Tokens as Tokens;
+
+/**
+ * Comprehensive I18n text domain fixer tool.
+ *
+ * This sniff can:
+ * - Add missing `text-domain`s.
+ * - Replace `text-domain`s based on an array of `old` values to a `new` value.
+ *
+ * Note: Without a user-defined configuration in a custom ruleset, this sniff will be ignored.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   1.2.0
+ */
+class I18nTextDomainFixerSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+		'CSS',
+	);
+
+	/**
+	 * Old text domain(s) to replace.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var string[]|string
+	 */
+	public $old_text_domain;
+
+	/**
+	 * New text domain.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var string
+	 */
+	public $new_text_domain = '';
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'i18nfixer';
+
+	/**
+	 * The WP Internationalization related functions to target for the replacements.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var array <string function name> => <int parameter position>
+	 */
+	protected $target_functions = array(
+		'load_textdomain'                        => 1,
+		'load_plugin_textdomain'                 => 1,
+		'load_muplugin_textdomain'               => 1,
+		'load_theme_textdomain'                  => 1,
+		'load_child_theme_textdomain'            => 1,
+		'unload_textdomain'                      => 1,
+
+		'__'                                     => 2,
+		'_e'                                     => 2,
+		'_x'                                     => 3,
+		'_ex'                                    => 3,
+		'_n'                                     => 4,
+		'_nx'                                    => 5,
+		'_n_noop'                                => 3,
+		'_nx_noop'                               => 4,
+		'translate_nooped_plural'                => 3,
+		'_c'                                     => 2, // Deprecated.
+
+		'esc_html__'                             => 2,
+		'esc_html_e'                             => 2,
+		'esc_html_x'                             => 3,
+		'esc_attr__'                             => 2,
+		'esc_attr_e'                             => 2,
+		'esc_attr_x'                             => 3,
+
+		'is_textdomain_loaded'                   => 1,
+		'get_translations_for_domain'            => 1,
+
+		// Shouldn't be used by plugins/themes.
+		'translate'                              => 2,
+		'translate_with_gettext_context'         => 3,
+
+		// WP private functions. Shouldn't be used by plugins/themes.
+		'_load_textdomain_just_in_time'          => 1,
+		'_get_path_to_translation_from_lang_dir' => 1,
+		'_get_path_to_translation'               => 1,
+	);
+
+	/**
+	 * Whether a valid new text-domain was found.
```

Comment:
Let's keep all instance (at least in comments) as two words: "text domain". Not a single word, not hyphenated.

Reply:
Should I also check for this in the `I18n` sniff and the documentation, i.e. changelog and such ?

Reply:
Yes please. Looking through the WP core functions (like [`__`](https://developer.wordpress.org/reference/functions/__/)), they all use two words for non-code instances (variables, etc.), so I think we should use the same too.

(Aside: "meta box" is often mistakenly written as "metabox" as well.)

Reply:
Well, I've adjusted it for this sniff in one of the new commits.

I suggest either one of us does a check for both `text[-]domain` as well as `meta[-]box` for the rest of the files and send that in as a separate PR to not pollute this PR with unrelated changes.